### ISSUE TITLE
403 error behind nginx

### ISSUE THREAD

**Issue by dmitriypavlov** [nirui/sshwifty#93]
Hi, could you please advice: im trying to deploy sshwifty behind dockerized nginx with config and always get 403 error

```
server {

    listen 80;
    server_name terminal.wombat.in.ua;
    server_tokens off;
    root /nowhere;
    return 301 https://terminal.wombat.in.ua$request_uri;
}

server {

    listen 443 ssl;
    server_name terminal.wombat.in.ua;
    server_tokens off;
    ssl_certificate /etc/nginx/ssl/live/wombat.in.ua/fullchain.pem;
    ssl_certificate_key /etc/nginx/ssl/live/wombat.in.ua/privkey.pem;
    ssl_protocols TLSv1.2 TLSv1.3;

    location / {
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "Upgrade";
        proxy_set_header Host $host;
        proxy_pass http://sshwifty:8182;
    }
}
```

**Comment by nirui** [comment]
Hi :D

Your Nginx configuration looked fine, assuming `http://sshwifty:8182` actually resolves to the host/container that runs Sshwifty and the `map` section was set correctly, of course. Maybe take look the logs on the Nginx container, see if there is anything related (or something should be there but missing, such as request logs).

On the Sshwifty side, the only thing I can think of that could give you a 403 error is the `"HostName"` (or `SSHWIFTY_HOSTNAME`) option in the configuration file. It checks if the client requested Hostname matches the configured one, and will show you a red error page if it doesn't. If that's the problem, simply clear the option to an empty value (which is the default) should resolve the issue.

So, to put it simply: If you saw a white 403 error page generated by Nginx, take look the log see if something is there; And if it was a red error page generated by Sshwifty, try clear the `"HostName"` option.

**Comment by dmitriypavlov** [comment]
**nirui**,  first of all I am very thankful for your detailed explanation and your time. I also do understand that the problem is in my setup, not your awesome software. Please forgive my ignorance, but your help is very appreciated.

Let me explode this by bits for the sake of understanding:

1. Nginx and sshfiwty are dockerized and both connected to `bridge` host network
2. `http://terminal.wombat.in.ua:8182` works just fine
3. When I try to connect via nginx proxy I always get 403 forbidden, while having in the nginx logs only:

`46.23.32.184 - - [08/May/2022:09:47:10 +0000] "GET /favicon.ico HTTP/1.1" 403 548 "https://terminal.wombat.in.ua/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.127 Safari/537.36" "-"`

4. In sshwifty logs there's simply nothing - I assume the request from nginx doesn't reach it at all

```
 Sshwifty Web SSH Client 
 Copyright (C) Ni Rui <ranqus@gmail.com>
 https://github.com/nirui/sshwifty
[INF] Sun, 08 May 2022 09:48:55 UTC Sshwifty: Initializing
[INF] Sun, 08 May 2022 09:48:55 UTC Sshwifty > Configuration > Redundant: Loading configuration from one of the default configuration files ...
[WRN] Sun, 08 May 2022 09:48:55 UTC Sshwifty > Configuration > Redundant: Unable to load configuration from "File": Configuration file was not specified. Also tried fallback files "/home/sshwifty/.config/sshwifty.conf.json", "/etc/sshwifty.conf.json", "/sshwifty.conf.json", but none of it was available
[INF] Sun, 08 May 2022 09:48:55 UTC Sshwifty > Configuration > Redundant: Loading configuration from environment variables ...
[INF] Sun, 08 May 2022 09:48:55 UTC Sshwifty > Server (0.0.0.0:8182): Serving

```

5. If I log into nginx container I can reach sshfifty manually:

```
root@8328970845f5:/# ping sshwifty
PING wombat.in.ua (185.26.98.98): 56 data bytes
64 bytes from 185.26.98.98: icmp_seq=0 ttl=64 time=0.087 ms
64 bytes from 185.26.98.98: icmp_seq=1 ttl=64 time=0.160 ms
64 bytes from 185.26.98.98: icmp_seq=2 ttl=64 time=0.166 ms
```

```
root@8328970845f5:/# telnet sshwifty 8182
Trying 185.26.98.98...
Connected to wombat.in.ua.
Escape character is '^]'.
```

The nginx config is quite simple and I see no reasons why it isn't working.
`SSHWIFTY_HOSTNAME` is left at default blank, I did not configure anything yet.

Your help is very appreciated!

**Comment by nirui** [comment]
> In sshwifty logs there's simply nothing - I assume the request from nginx doesn't reach it at all

I was expecting this actually. Usually when Nginx cannot reach the upstream, it should return a 500 code, such as 503, not 403.

Earlier, I've tested the configuration that you provided in your first comment, and I can confirm it works. The only thing that I have to add was:

    map $http_upgrade $connection_upgrade {
        default upgrade;
        '' close;
    }

which is the `map` setting that I mentioned above. I assumed this was set correctly, because even if it didn't, it should not cause a 403 error. 

Now, I'm actually as confused as you are :D

Anyhow, now we know the problem was not caused neither by Sshwifty nor your configuration. The next stop could be a closer look at `nginx.conf` or other global settings. Particularly, to see if there is a default server (usually located at `/etc/nginx/conf.d/default.conf`), or a "catch all" server which should look like:

    server {
        server_name _;
        ......
    }

notice the `server_name` is a `_`.

**Comment by dmitriypavlov** [comment]
I am very sorry! I found my mistake. I am so stupid - the conf file for nginx sshwifty server had no extension and was not included from the nginx.conf, also you were right - there was a default 80/443 server which catched my invalid subdomain.

Thank you so much for your patience. Sorry for taking your time!

**Comment by nirui** [comment]
It's fine, cheers :D
